plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/util/FeatureDevService.kt

```diff
@@ -138,6 +140,10 @@ class FeatureDevService(val proxyClient: FeatureDevClient, val project: Project)
                     (e is ThrottlingException && e.message?.contains("limit for number of iterations on an implementation plan") == true)
                 ) {
                     throw PlanIterationLimitError(message("amazonqFeatureDev.approach_gen.iteration_limit.error_text"), e.cause)
+                } else if (e is StreamingValidationException && e.message?.contains("repo size is exceeding the limits") == true) {
+                    throw ContentLengthError(message("amazonqFeatureDev.content_length.error_text"), e.cause)
+                } else if (e is StreamingValidationException && e.message?.contains("zipped file is corrupted") == true) {
+                    throw ZipFileError("The zip file is corrupted", e.cause)
```

Comment:
Should we propose an action to take for the customers? To me it feels that, saying just `the zip file is corrupted` can be very confusing since the user doesn't even know what file we're talking about here.

Reply:
Actually, there is not action to be performed by user; we are asking them to just `SendFeedback` without any Retry/action. I had similar comment for VSCode change: https://github.com/aws/aws-toolkit-vscode/pull/5245#discussion_r1658507067 
Will confirm with @gasolima and update.

Reply:
This only can happen if we've a bug in our toolkit code, and in as many retries the user can do will not help. We will need to update the toolkit version. So from customer perspective they don't have anything to do.